### DISCUSSION TITLE
Screengrab doesn't start anymore with qt 5.15.4

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#2243] [General]
```
$ screengrab 
Cannot mix incompatible Qt library (5.15.3) with this library (5.15.4)
Aborted (core dumped)
```

Recompiling https://github.com/lxqt/libqtxdg/ and screengrab fixed this - other apps doesn't seem to be affected.

**Comment by tsujan** [comment]
I encountered the same problem, but I think the Arch package libqtxdg 3.9.1-2 is already recompiled; isn't it?

BTW, recompilation of `screengrab` wasn't required, as far as I remember.

EDIT: I mean, the problem may happen only to those who make their own packages,

**Reply by yan12125** [reply]
> Arch package libqtxdg 3.9.1-2 is already recompiled; isn't it?

Yep https://github.com/archlinux/svntogit-community/commit/9f8f2e0ed6e4a62c501d541c9243eaeb54c827a8

Both screengrab and patched xdg-utils works for me without recompilation, by the way. On this machine I install everything from Arch repositories instead of git. Probably you've installed something from git and overwritten Arch packages?

**Reply by tsujan** [reply]
> Both screengrab and patched xdg-utils works for me without recompilation,

Yes, here too.

**Comment by tsujan** [comment]
> other apps doesn't seem to be affected.

Having patched `xdg-utils` (→ https://github.com/lxqt/libqtxdg/issues/264), I couldn't open any link ;) `qtxdg-tools` was also affected.

Such problems can happen to programs that include Qt's private headers, whether in LXQt or in KDE.

**Comment by tsujan** [comment]
Happened again with Qt 5.15.5+kde+r163

**Comment by darkshram** [comment]
These are the current runtime dependencies for screengrab in Fedora/ALDOS (one advantage of having a package manager with capabilities for tracking dependencies):

```
libICE.so.6()(64bit)
libKF5WindowSystem.so.5()(64bit)
libQt5Core.so.5()(64bit)
libQt5Core.so.5(Qt_5)(64bit)
libQt5Core.so.5(Qt_5.15)(64bit)
libQt5DBus.so.5()(64bit)
libQt5DBus.so.5(Qt_5)(64bit)
libQt5Gui.so.5()(64bit)
libQt5Gui.so.5(Qt_5)(64bit)
libQt5Network.so.5()(64bit)
libQt5Network.so.5(Qt_5)(64bit)
libQt5Svg.so.5()(64bit)
libQt5Widgets.so.5()(64bit)
libQt5Widgets.so.5(Qt_5)(64bit)
libQt5X11Extras.so.5()(64bit)
libQt5X11Extras.so.5(Qt_5)(64bit)
libQt5Xdg.so.3()(64bit)
libQt5XdgIconLoader.so.3()(64bit)
libQt5Xml.so.5()(64bit)
libX11-xcb.so.1()(64bit)
libX11.so.6()(64bit)
libXext.so.6()(64bit)
libc.so.6()(64bit)
libc.so.6(GLIBC_2.14)(64bit)
libc.so.6(GLIBC_2.2.5)(64bit)
libc.so.6(GLIBC_2.4)(64bit)
libextedit.so.1.0.0()(64bit)
libgcc_s.so.1()(64bit)
libgcc_s.so.1(GCC_3.0)(64bit)
libgio-2.0.so.0()(64bit)
libglib-2.0.so.0()(64bit)
libgobject-2.0.so.0()(64bit)
libm.so.6()(64bit)
libstdc++.so.6()(64bit)
libstdc++.so.6(CXXABI_1.3)(64bit)
libstdc++.so.6(CXXABI_1.3.9)(64bit)
libstdc++.so.6(GLIBCXX_3.4)(64bit)
libxcb-shape.so.0()(64bit)
libxcb-xfixes.so.0()(64bit)
libxcb.so.1()(64bit)
```

The relevant Qt5 related dependency would be **libqtxdg**. So, this wold be the relevant LXQt package for screengrab to rebuild after all Qt5 suite.

I will share the list of most common packages using private headers that need to be rebuild with every Qt5 release. Package names are from Fedora and ALDOS. List was obtained from ALDOS using `dnf repoquery --whatrequires` for **all** Qt5 private libraries. Short story: every time a new Qt5 release show up, all the following packages need to be rebuilt. Hope this information helps solve the issue. The relevant packages for LXQt required to be rebuilt are **libqtxdg**, **libfm-qt** and **lxqt-qtplugin**, in that order.

```
analitza
appmenu-qt5
calibre
fcitx-qt5
gpxsee
haruna
kf5-akonadi-server
kf5-kxmlgui
kmymoney-libs
kwin-libs
kwin-wayland
layer-shell-qt
libfm-qt
libqtxdg
linphone-desktop
liquidshell
lumina-desktop
lxqt-qtplugin
nextcloud-client
plasma-integration
pyotherside
python3-pyqtchart
python3-pyside2
python3-qt5
python3-qt5-base
qgnomeplatform
qt5ct
qt5-designer
qt5-doctools
qt5gtk2
qt5-qtstyleplugins
qt5-qtcanvas3d
qt5-qtwebkit
qtav-qml-module
qt-creator
qtcurve-qt5
skrooge-libs
texmaker
xdg-desktop-portal-kde
```
